test/unit/gateways/global_collect_test.rb

```diff
@@ -71,6 +86,30 @@ def test_successful_purchase_with_requires_approval_true
     end.respond_with(successful_authorize_response, successful_capture_response)
   end
 
+  def test_purchase_request_with_google_pay
+    stub_comms do
+      @gateway.purchase(@accepted_amount, @google_pay_network_token)
+    end.check_request(skip_response: true) do |_endpoint, data, _headers|
+      assert_equal '320', JSON.parse(data)['mobilePaymentMethodSpecificInput']['paymentProductId'] if JSON.parse(data)['mobilePaymentMethodSpecificInput']
+    end
+  end
+
+  def test_purchase_request_with_apple_pay
```

Comment:
💬  Comment:

The is no value in the testing purchase and authorize because purchase basically makes a delegate request to authorize and doesn't modify any params so at the end you are testing the same code twice.

Reply:
You are right. Test purchase tests an authorize and tests a capture.